packages/design/src/ButtonOutlined/ButtonOutlined.tsx

```diff
@@ -14,23 +14,44 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-import React from 'react';
+import React, { ComponentPropsWithoutRef, ElementType } from 'react';
 import styled from 'styled-components';
 
 import { space, width } from 'design/system';
-import defaultTheme from 'design/theme';
 
-const ButtonOutlined = (
-  { children, setRef, ...props } = { setRef: undefined }
-) => {
+import type { SpaceProps, WidthProps } from 'design/system';
+
+import type { PropsWithTheme } from 'design/theme';
+
+interface ButtonOutlinedBaseProps<T extends ElementType> {
+  setRef?: React.Ref<HTMLButtonElement>;
```

Comment:
I think more idiomatic way would be to use `React.forwardRef` https://reactjs.org/docs/forwarding-refs.html. But thats not connected to this PR